readthedocs/projects/models.py

```diff
@@ -144,11 +144,19 @@ class AddonsConfig(TimeStampedModel):
     Everything is enabled by default.
     """
 
-    DOC_DIFF_DEFAULT_ROOT_SELECTOR = "[role=main]"
-    LINKPREVIEWS_DEFAULT_ROOT_SELECTOR = "[role=main] a.internal"
-    LINKPREVIEWS_DOCTOOL_NAME_CHOICES = (
-        ("sphinx", "Sphinx"),
-        ("other", "Other"),
+    DEFAULT_ROOT_SELECTOR = ", ".join(
+        [
+            "main > div > div.md-content",  # Material for MkDocs
+            "article#main",  # Docsify
+            "article",  # Pelican, Docusaurus, Antora, Jekyll
+            "div#content",  # AsciiDoctor
+            "[role=main]",
+            "main",
+            "div.body",
+            "div.document",
+            "h1",  # Very generic fallback
+            "body",  # Fallback
```

Comment:
Yea, it definitely seems like the client can be smarter about the ordering/logic around selectors, even expanding beyond just basic CSS selectors. I'm not sure there's much value in passing a default from the backend.

Reply:
Do we think we'll need this additional logic in the search indexing? Maybe we should focus on polishing Addons first and come back to see if there is a way to reuse some of this in search indexing afterwards?

Reply:
This makes ton of sense to me. This was my original proposal: "use a different CSS selector based on the documentation tool", but I understood you preferred to use a more generic CSS selector for all of them.

I'm happy to build the auto-detection on the client side. We already have some of this implemented for EthicalAds, so we can probably re-use/re-write that logic eventually.

I found issues with the search tests when using the generic CSS selector, so I removed it from there and kept the original one for search. However, _ideally_, I'd like to share the same CSS selector all across our platform because we won't be indexing the right content for some documentation tools if we use the current logic in search/filetreediff (backend features).

Reply:
Yea, it would be nice to have the same logic in both, so it's obvious if it works in 1 place, it works in all of them as a user. It might be hard to keep them in sync, but if it gets really complex, we could investigate running node on the servers to actually parse an HTML doc as a worst case.